js/quote-generator.js

```diff
@@ -0,0 +1,55 @@
+const randomBtn = document.querySelector("#random-btn");
+const allAuthorsBtn = document.querySelector("#all-authors-btn");
+const quote = document.querySelector(".quote");
+let quoteAuthor = document.querySelector(".quote-author");
+let quoteGenre = document.querySelector(".quote-genre");
+const allAuthorsContainer = document.querySelector(".all-authors-container");
+const gridContainer = document.querySelector(".grid-container");
+const topButtons = document.querySelector(".top-buttons");
+const footer = document.querySelector("footer");
+
+async function getRandomQuote() {
+  const promise = await fetch("https://api.quotable.io/random");
+  const result = await promise.json();
+  quote.textContent = result.content;
+  quoteAuthor.textContent = result.author;
+  quoteGenre.textContent = result.tags.join(", ");
+}
+
+randomBtn.addEventListener("click", getRandomQuote);
+getRandomQuote();
+
+async function getAllAuthors() {
+  const promise = await fetch(
+    `https://api.quotable.io/authors?sortBy=quoteCount&order=desc&limit=10`
+  );
+  const result = await promise.json();
+  let count = 1;
```

Comment:
You don't need this variable to implement counting of authors

Consider using `entries` method that will create an iterator of you 

Iterate over this iterator like `for (let authorEntry of result.results.entries()) {}`

Each item is a tuple where first element is an index and the second element is an author object

Then you can do something like this
```
const [idx, author] = authorEntry;
const authorNumber = idx + 1; // it's your count
```